geoapps_utils/driver/data.py

```diff
@@ -45,6 +45,24 @@ class BaseData(BaseModel):
     monitoring_directory: str | Path | None = None
     _input_file: InputFile | None = None
 
+    @staticmethod
+    def get_aliases_value(info, field, data: dict[str, Any]) -> Any:
```

Comment:
Just spitballing here.. I wonder if we could rewrite the `collect_input_from_dict` method to construct all the nested pydantic BaseModels passing in all the data (since we have `allow_extra=True`) and then just construct the update dictionary by calling the `model_dump()` (ignores extra fields).  This would use the internal pydantic alias handling, so we don't need a to reinvent the wheel here.  It also avoids the recursion, so it would be more efficient.